### PR TITLE
stage1: add @hasField() built-in

### PR DESCRIPTION
**Pull request by shawnl**
This was quite straight-forward

Closes: #1439

### CODE REVIEW

**Comment by shawnl** [comment]
It doesn't work on imports.

**Comment by emekoi** [comment]
shouldn't it be `@hasField(comptime T: type, comptime name: []const u8) bool`? why must name be mutable?

**Comment by shawnl** [comment]
@emekoi your right. That is the correct documentation of what the code already does.

File: src/all_types.hpp [hunk 1]
```diff
@@ -3229,6 +3231,14 @@ struct IrInstructionTypeInfo {
     IrInstruction *type_value;
 };
 
+struct IrInstructionHasField {
+    IrInstruction base;
+
+    IrInstruction *container_type;
+    Buf *field_name_buffer;
```
Comment (andrewrk) [inline src/all_types.hpp]: I don't think the existence of this field is correct. I think you can follow this pattern: 269a53b6afb052ec09f7b26db68219cfeaa0460e

Comment (shawnl) [reply src/all_types.hpp]: I was following another existing pattern.

**Comment by andrewrk** [comment]
Thanks, and please have a look at the fixups commit: df11512f85e5228ee2393e9dc5fc42294ecba1c9

**Pull request merged**